packages/ui/components/overlays/test/OverlayController.test.js

```diff
@@ -1764,4 +1764,20 @@ describe('OverlayController', () => {
       );
     });
   });
+
+  describe('run _keepBodySize only with global placement', () => {
+    /**
+     * @type {OverlayController}
+     */
+    const overlayControllerLocal = new OverlayController({
+      ...withLocalTestConfig(),
+    });
+
+    it('should not run with local placement', async () => {
+      await overlayControllerLocal.show();
+
+      expect(overlayControllerLocal.__bodyMarginRightInline).to.equal(undefined);
+      expect(overlayControllerLocal.__bodyMarginRight).to.equal(undefined);
+    });
```

Comment:
Looks great, maybe you can also test here for the `preventsScroll: true` case...

```
      overlayControllerLocal.updateConfig({ preventsScroll: true }); 

      expect(overlayControllerLocal.__bodyMarginRightInline).to.not.equal(undefined);
      expect(overlayControllerLocal.__bodyMarginRight).to.not.equal(undefined);

```